### PR TITLE
Linux 6.1.6

### PR DESCRIPTION
**Pull request by 13r0ck**
- [x] Test pylmouth on Mira with Nvidia before general kernel QA. 1b088b00d9f31dcecc5751e9c26007b12706f4f5 no longer cherry-picks cleanly. So hopefully it has been fixed correctly, otherwise more work will be needed.

This will be the first Pop!_OS kernel with rust in it! :crab:

### CODE REVIEW

**Comment by 13r0ck** [comment]
This has now also pulled in #215. So this will also not have that vulnerability,

**Comment by jglathe** [comment]
Feedback: I could build it on my build machine (ubuntu 22.10 server as VM).  Had to clean up a little after the first run that failed on dkms, but was just build debris still lying around, `git clean -df` took care of it. It is running happily on my vintage T500, will test it on a Ryzen box with NVidia card and 2 screens, too.

**Comment by 13r0ck** [comment]
@jglathe awesome! Thank you!

**Review by leviport — CHANGES_REQUESTED** [review]
This breaks suspend on pang10 and pang11. Related fix: https://gitlab.freedesktop.org/superm1/linux/-/commit/29c666b4184a41bb07acd24551e4cd53276340b2

**Comment by 13r0ck** [comment]
Cherry-picked that fix, so that shouldn't be an issue anymore

**Comment by leviport** [comment]
Unfortunately that didn't fix suspend on pang10

File: drivers/gpio/gpiolib-acpi.c
Comment (jackpot51) [inline drivers/gpio/gpiolib-acpi.c:1623]: @13r0ck this over-matches, use product version, one for pang10 and another for pang11, or find the latest version of the patch.

**Comment by 13r0ck** [comment]
5ee6a55e521dc94bc2090d467b6281461f79220f should fix the pang10/12 suspend issue

**Comment by leviport** [comment]
Yep, looks like that fixed pang11! :tada:

**Comment by leviport** [comment]
Looks like `bcmwl-kernel-source` is throwing some DKMS errors when I try to install it. Here is the log: 
[make.log](https://github.com/pop-os/linux/files/10523474/make.log)

Guessing https://github.com/pop-os/bcmwl/ just needs an update as well?

**Comment by 13r0ck** [comment]
https://github.com/pop-os/bcmwl/tree/bcmwl-rebase fixes that for me

**Comment by leviport** [comment]
Yep, that fixes it for me as well

**Comment by XV-02** [comment]
@13r0ck can you elaborate on the Mira/Nvidia/Plymouth issue? 

On the first few attempts, I struggled to see a Mira/Nvidia system progress beyond entering the decryption password on boot with this kernel, which is a worry.

I've attached a log (ripped from journalctl) from the last failed attempt to boot with this Kernel PR present. 
[Failure.txt](https://github.com/pop-os/linux/files/10539071/Failure.txt)

I'm in the process of seeing whether I can find some way to induce a boot failure now I've had the system boot successfully.

**Comment by XV-02** [comment]
Previous log was after a failure to boot Old_Kern after adding this PR. This is after a failure to boot 6.1.6:
[Failure2.txt](https://github.com/pop-os/linux/files/10539234/Failure2.txt)

**Comment by XV-02** [comment]
Two back to back boot attempts, one good and one bad. They diverge at line 762 (I believe they are _literally_ identical for the first 761 line, time excepted). Not sure the implications there of. 

[good.txt](https://github.com/pop-os/linux/files/10539454/good.txt)
[bad.txt](https://github.com/pop-os/linux/files/10539455/bad.txt)

**Comment by jglathe** [comment]
nvidia driver balking

**Comment by XV-02** [comment]
At least on DevOne, things are looking good. 

[DevOneLinux6-1-6.txt](https://github.com/pop-os/linux/files/10549734/DevOneLinux6-1-6.txt)

But, Mira is still misbehaving. The fact that it doesn't fail to move from Plymouth decrypt to login every time is odd to me, and guidance on desired debugging would be helpful.

**Comment by 13r0ck** [comment]
@XV-02 I found a new upstream patch (f8619fafc844d77bbbf26d032453cd37abc685cc) and added it here, please let me know if that works

**Comment by 13r0ck** [comment]
From my personal testing 20cfe481ed013f15293c564f51ceb276910c5588 seems to work much better

**Comment by dvdsk** [comment]
No idea if this is the right place to report this but this kernel misses the zfs module. (though maybe that is supposed to be the case and the module is added later)

I installed it via `sudo apt-manage add popdev:linux-6.1.6`

**Comment by XV-02** [comment]
I'm seeing a regression in behaviour on Oryp10:

On 6.0.12, booting with lid closed and HDMI external monitor, I can decrypt the drive and log in to my user session. At that point, the machine enter sleep, and tapping the space bar wake the machine and you arrive at your user session.

On 6.1.6, booting with lid closed and HDMI external monitor, I can decrypt the drive, and at that point, the process stalls. The external monitor is blank, and log in is never reached. Opening the lid displays a Plymouth message reporting that the drive was decrypted, but doesn't proceed to login. 

Something is responsive, as tapping the spacebar on an external keyboard wake the external display, but it remains otherwise blank.

**Review by XV-02 — DISMISSED** [review]
The size of the last revision is small, and the specific implications have been tested across a range of hardware (Dev One, Lemp11, Mira B4, Oryp10, Mega R1)

Outside that, things are looking good, the only caveat is that **this must be published with [Nvidia 525.85.05](https://github.com/pop-os/nvidia-graphics-drivers/pull/176).**

**Comment by 13r0ck** [comment]
The last thing we are waiting for is the zfs rebase https://github.com/pop-os/zfs-linux/pull/13 (thanks @dvdsk)

**Review by jackpot51 — CHANGES_REQUESTED** [review]
Just want to block merge until ZFS change is approved and merged. Everything else looks good, including the latest patch to fix NVIDIA framebuffer issues

**Comment by n3m0-22** [comment]
ZFS is good to go.

**Comment by 13r0ck** [comment]
@jackpot51 zfs-linux was approved and merged

**Comment by leviport** [comment]
It appears that this breaks DP MST on Dev One, just like in https://github.com/pop-os/linux/issues/187

**Comment by 13r0ck** [comment]
> It appears that this breaks DP MST on Dev One, just like in https://github.com/pop-os/linux/issues/187

Works on my machine

**Comment by leviport** [comment]
Alright, I've got a better idea of what's going on, and I've eliminated Mesa 22.3.4 from the equation. With this kernel installed, MST breaks after suspend and resume on Dev One. So from a fresh boot, everything is working fine.

After suspending and resuming with this 6.1.6 on a fresh and updated install, MST stops working when I plug the USB-C cable in. Display settings shows both monitors I have connected, but the upstream one says "No signal" and goes back to sleep, while the downstream one never even wakes up.

When booting back into oldkern (6.0.12), both monitors work fine after suspending and resuming, then connecting the USB-C cable. I've replicated this on my own Dev One as well.

**Comment by lolcosmo** [comment]
Could you enable MGLRU by default?

The kernel config option is CONFIG_LRU_GEN_ENABLED

https://www.phoronix.com/news/MGLRU-Reaches-mm-stable

**Comment by 13r0ck** [comment]
@lolcosmo good find!. Currently too much of our QA has been done already on this kernel release. But I will enable it on our 6.2 series kernel and see if it passes QA.

Until the next release you could build your own kernel with that enabled
```
git clone https://github.com/pop-os/linux
cd linux
vim debian.master/config/config.commonubuntu <- this is where to enable MGLRU
sudo apt build-dep linux-system76 -y
dpkg-build-package -us -uc -b
cd ..
sudo dpkg -i *.deb
reboot
```

**Comment by 13r0ck** [comment]
@leviport I cherry-picked eeda91289948ab5381847ff5d52279570d1eaeac from 6.1.9, that should fix the MST bug on DevOne

**Review by jackpot51 — APPROVED** [review]

**Comment by leviport** [comment]
Sorry, but I don't think this fixes it. If anything, it seems worse now. Instead of just not activating either of the displays, now it sometimes makes the built-in display go black with only the mouse cursor shown, and act locked up (can't move the mouse cursor) until I unplug the USB-C cable. I tried to make these things happen while booted into oldkern (6.0.12) on the same install, but it just kept working as expected between reboots and suspend/resume cycles.

I'll do a reinstall and try again soon, just in case it was some kind of fluke.

**Comment by 13r0ck** [comment]
Closed in favor of #222

**Pull request closed without merging**
